sdk/core/Azure.Core/src/Serialization/AzureJsonSerializer.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Reflection;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+#pragma warning disable CS1591, SA1649, SA1402
+
+namespace Azure.Core.Serialization
+{
+    // For libraries that operate on opaque binary data (event hubs)
+    public abstract class AzureObjectSerializerFactory
+    {
+        public abstract AzureObjectSerializer Create(Type type);
+        // Async to allow to support scenarios where schema is retrieved from somewhere
+        public abstract ValueTask<AzureObjectSerializer> CreateAsync(Type type);
+    }
+
+    // For libraries that need to have json specific features (tables, search)
+    public abstract class AzureJsonSerializerFactory: AzureObjectSerializerFactory
+    {
+        public abstract AzureJsonSerializer CreateJsonSerializer(Type type);
+        public abstract ValueTask<AzureObjectSerializer> CreateJsonSerializerAsync(Type type);
+    }
+
+    // Represents a serializer/deserializer from any byte stream to an object
+    public abstract class AzureObjectSerializer
+    {
+        public abstract void Serialize(object o, Stream stream);
+        public abstract ValueTask SerializeAsync(object o, Stream stream);
+
+        public abstract object Deserialize(Stream stream);
+        public abstract ValueTask<object> DeserializeAsync(Stream stream);
+    }
+
+    // Represents a JSON specific serializer
+    // Allows adding more JSON specific/optimized methods
+    public abstract class AzureJsonSerializer: AzureObjectSerializer
+    {
+        // The following methods are experiments on what JSON specific methods are
+
+        // If this serializer supports reading from Utf8JsonReader but want to add some customizations on top of that
+        public abstract bool SupportsUtf8Json { get; }
+        public abstract void Serialize(object o, Utf8JsonWriter stream);
```

Comment:
I don't think we can have specific serialization technology types in the abstraction APIs.

Reply:
This is just an idea but  I think it's settled that we would always use System.Text.Json by default and have a dependency on it so if customers deserializer can have a fast path reading from already pre-parsed json what's wrong with that?